lib/with_model.rb

```diff
@@ -7,36 +7,46 @@
 
 module WithModel
   # @param [Symbol] name The constant name to assign the model class to.
-  # @param scope Passed to `before`/`after` in the test context.
+  # @param scope Passed to `before`/`after` in the test context. RSpec only.
   # @param options Passed to {WithModel::Model#initialize}.
   # @param block Yielded an instance of {WithModel::Model::DSL}.
   def with_model(name, scope: nil, **options, &block)
     model = Model.new name, **options
     dsl = Model::DSL.new model
     dsl.instance_exec(&block) if block
 
-    before(*scope) do
-      model.create
-    end
-
-    after(*scope) do
-      model.destroy
-    end
+    setup_object(model, scope)
   end
 
   # @param [Symbol] name The table name to create.
-  # @param scope Passed to `before`/`after` in the test context.
+  # @param scope Passed to `before`/`after` in the test context. Rspec only.
   # @param options Passed to {WithModel::Table#initialize}.
   # @param block Passed to {WithModel::Table#initialize} (like {WithModel::Model::DSL#table}).
   def with_table(name, scope: nil, **options, &block)
     table = Table.new name, options, &block
 
-    before(*scope) do
-      table.create
-    end
+    setup_object(table, scope)
+  end
+
+  private
+
+  # @param [Object] object The new model object instance to create
+  # @param scope Passed to `before`/`after` in the test context. Rspec only.
+  def setup_object(object, scope = nil)
+    if Module.const_defined? 'RSpec'
+      before(*scope) do
+        object.create
+      end
+
+      after(*scope) do
+        object.destroy
+      end
+    elsif Module.const_defined? 'Minitest'
```

Comment:
Won't this also catch Minitest::Spec?

Reply:
Yeah, good point - very likely, just pure luck I put the RSpec one first. Can you think of a better way to determine which test suite is executing?

Reply:
I can't.  May be able to check for `Minitest::Spec` first, then fall back to `Minitest` but even if that happens to work now it seems like this is getting too far into the internals of Minitest.

Could add a `test_suite` parameter to `with_model` and default to `:rspec`.  

Could also encourage overwriting `setup_object`.  That way the developer can use whatever callbacks with whatever parameters, or even something like:
```
def setup_object(object, **args)
  object.create
  yield
  object.destroy
end
```

Maybe the intention is to use the DSL API instead in this case though?